pkg/client/rules.go

```diff
@@ -12,15 +12,30 @@ import (
 	"github.com/grafana/cortex-tools/pkg/rules/rwrulefmt"
 )
 
+const (
+	rulerAPIPath  = "/api/v1/rules"
+	legacyAPIPath = "/api/prom/rules"
+)
+
 // CreateRuleGroup creates a new rule group
 func (r *CortexClient) CreateRuleGroup(ctx context.Context, namespace string, rg rwrulefmt.RuleGroup) error {
 	payload, err := yaml.Marshal(&rg)
 	if err != nil {
 		return err
 	}
 
+	path := rulerAPIPath
+	if r.legacy {
+		path = legacyAPIPath
+	}
 	escapedNamespace := url.PathEscape(namespace)
-	res, err := r.doRequest("/api/prom/rules/"+escapedNamespace, "POST", payload)
+	path += "/" + escapedNamespace
```

Comment:
This can be simplified if the path prefix is stored as a string in the client struct.

Reply:
I agree. I'll simplify it.